### ISSUE TITLE
Recurrent on voc2007+2012

### ISSUE THREAD

**Issue by Tangzy7** [yhenon/keras-frcnn#55]
I wanted to achieve the benchmark of the paper, which is a 70% mAP on voc2007+voc2012. I used this  repository to train on voc2007+2012 and test on voc2007_test, but I only got a mAP of 61%. Did anyone recurrent the result successfully?

**Comment by yhenon** [comment]
I've shared the weights I obtained from training with this repo here: #23
A map of 60 is quite low, I'd have to know how many epochs/iterations did you train for, the learning rate, etc

**Comment by Tangzy7** [comment]
@yhenon Hi yhenon, thanks for your reply and the weight file! I still have 2 problems.
1. I try your config and hdf5 file to evaluate with measure_map.py and I get a mAP of 80%  . And the benchmark is 73%. I just wondered the way you calculate mAP was different from the [paper's.](https://github.com/rbgirshick/py-faster-rcnn/blob/master/lib/datasets/voc_eval.py) I used the official way to test your hdf5 file and got a mAP of 58%

2. I mentioned that the way you calculate IOU didn't follow definition of union area. The union area you calculate is bigger that actual one. It happen in "def union(au, bu)" and "non_max_suppression_fast". Are you meant to do in this way? Although this calculation won't affect too much, it still have a small influence on performance

def union(au, bu):
	x = min(au[0], bu[0])
	y = min(au[1], bu[1])
	w = max(au[2], bu[2]) - x
	h = max(au[3], bu[3]) - y
	return x, y, w, h

**Comment by yhenon** [comment]
Thanks for pointing out that bug - I'm still working things out!

**Comment by yhenon** [comment]
@Tangzy7 any chance you could share the code you used to test the official way? Would be a great help

**Comment by Tangzy7** [comment]
@yhenon I used this official script 
https://github.com/rbgirshick/py-faster-rcnn/blob/master/lib/datasets/voc_eval.py

**Comment by shahruk10** [comment]
@yhenon  hi, does fixing this bug improve the results of training significantly? I ask because I just completed training a large dataset of my own of small target objects (50 ~ 200 px) and got a mean RPN bounding box overlap with GT = 8.17.
